.github/workflows/build.yml

```diff
@@ -26,14 +26,10 @@ jobs:
             *.cache-from=type=gha,scope=build
             *.cache-to=type=gha,scope=build,mode=max
 
-  validate:
```

Comment:
Actually I think we could keep the matrix in the `validate` job to reduce workflow overhead and also keep required GitHub checks valid:

```yaml
  validate:
    runs-on: ubuntu-20.04
    strategy:
      fail-fast: false
      matrix:
        target:
          - htmlproofer
          - mdl
    steps:
      -
        name: Checkout
        uses: actions/checkout@v3
      -
        name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v2
      -
        name: Validate
        uses: docker/bake-action@v2
        with:
          targets: ${{ matrix.target }}
          set: |
            *.cache-to=type=gha,scope=validate-${{ matrix.target }},mode=max
            *.cache-from=type=gha,scope=validate-${{ matrix.target }}
            *.cache-from=type=gha,scope=build
      -
        name: Annotate
        uses: actions/github-script@v6
        with:
          script: |
            const fs = require('fs');
            const results = fs.readFileSync('lint/results', 'utf-8');
            console.log(results);

            if ("${{ matrix.target }}" == "htmlproofer") {
              const re = /^- (.+)\n  \*  (.+) \(line (\d+)\)\n(.+)$/gm;
              while (true) {
                const result = re.exec(results);
                if (result === null) {
                  break;
                }
                core.error(`${result[2]}\n${result[4]}`, {
                  title: 'Link check failed',
                  // file: result[1],
                  // startLine: result[3],
                })
              }
              if (results.length > 0) {
                process.exitCode = 1
              }
            } else if ("${{ matrix.target }}" == "mdl") {
              for (const result of results) {
                core.error(result.description, {
                  title: result.rule + (result.aliases.length > 0 ? ` (${result.aliases[0]})` : ``),
                  file: result.filename,
                  startLine: result.line,
                })
              }
              if (results.length > 0) {
                process.exitCode = 1
              }
            }
```

Reply:
Agreed, that feels significantly better to me :+1: